src/main/java/org/spongepowered/api/item/inventory/type/ChestInventory.java

```diff
@@ -22,55 +22,33 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.item.inventory.custom;
+package org.spongepowered.api.item.inventory.type;
 
 import org.spongepowered.api.Sponge;
-import org.spongepowered.api.item.inventory.type.OrderedInventory;
-import org.spongepowered.api.text.translation.Translation;
-import org.spongepowered.api.util.Identifiable;
-import org.spongepowered.api.util.ResettableBuilder;
+import org.spongepowered.api.item.inventory.custom.CustomInventoryBuilder;
 
 /**
- * A CustomInventory is an inventory that has no backing in the world that is
- * created by plugins.
+ * A vanilla ChestInventory with a Grid of width 9
  */
-public interface CustomInventory extends OrderedInventory, Identifiable {
+public interface ChestInventory extends GridInventory {
 
     /**
-     * Creates a new {@link Builder} to build a {@link CustomInventory}.
+     * Creates a new {@link Builder} to build a {@link ChestInventory}.
      *
      * @return The new builder
      */
     static Builder builder() {
         return Sponge.getRegistry().createBuilder(Builder.class);
     }
 
-
-    interface Builder extends ResettableBuilder<CustomInventory, Builder> {
-        /**
-         * Sets the title of the custom inventory, viewable by players looking
-         * at the inventory.
-         *
-         * @param name The title
-         * @return fluent pattern
-         */
-        Builder name(Translation name);
+    interface Builder extends CustomInventoryBuilder<ChestInventory, Builder> {
 
         /**
          * Sets the size of the custom inventory.
          *
          * @param size The size
          * @return fluent pattern
          */
-        Builder size(int size);
-
-        /**
-         * Builds the inventory instance.
-         *
-         * @return A new custom inventory
-         */
-        CustomInventory build();
-
-        // TODO add persistence methods
+        ChestInventory.Builder rows(int size);
```

Comment:
tbh rows is more understandable as in a chest GUI the number changes the number of rows that are shown.